components/GameScreen.vue

```diff
@@ -106,6 +109,120 @@ watch(gameState, async (newState, prevState) => {
+      if (!appRef.value) {
+        return;
+      }
+
+      isZoomingOut = true;
+      startZoomTime = performance.now();
+      startMousePos = mousePos;
+      requestAnimationFrame(animateZoomOut);
+    }
+
+    function animateZoomOut(currentTime: number) {
+      if (!appRef.value || !startZoomTime) {
+        return;
+      };
+
+      const elapsedTime = currentTime - startZoomTime;
+      const progress = Math.min(elapsedTime / zoomDuration, 1);
+      const newScale = minZoom + (appRef.value.stage.scale.x - minZoom) * (1 - progress);
+
+      const worldPos = {
+        x: (startMousePos.x - appRef.value.stage.position.x) / appRef.value.stage.scale.x,
+        y: (startMousePos.y - appRef.value.stage.position.y) / appRef.value.stage.scale.y,
+      };
+
+      appRef.value.stage.scale.set(newScale);
+
+      const newScreenPos = {
+        x: worldPos.x * newScale + appRef.value.stage.position.x,
+        y: worldPos.y * newScale + appRef.value.stage.position.y,
+      };
+
+      appRef.value.stage.position.set(
+        Math.min(0, Math.max(appRef.value.stage.position.x - (newScreenPos.x - startMousePos.x), appRef.value.screen.width - appRef.value.screen.width * newScale)),
+        Math.min(0, Math.max(appRef.value.stage.position.y - (newScreenPos.y - startMousePos.y), appRef.value.screen.height - appRef.value.screen.height * newScale)),
+      );
+      if (progress < 1) {
+        requestAnimationFrame(animateZoomOut);
+      } else {
+        isZoomingOut = false;
+      }
+    }
+
+    // Mouse wheel event listener
+    canvas.value?.addEventListener("wheel", (event) => {
+      event.preventDefault();
+      const mousePos = { x: event.offsetX, y: event.offsetY };
+      if (event.deltaY > 0) {
+        // Zoom out when scrolling down
+        if (!isZoomingOut) {
+          smoothZoomOut(mousePos);
+        }
+      } else {
+        // Existing zoom-in functionality
+        const zoomFactor = event.deltaY * -zoomSpeed;
+        const newScale = Math.max(minZoom, Math.min(maxZoom, app.stage.scale.x + zoomFactor));
+
+        const worldPos = {
+          x: (mousePos.x - app.stage.position.x) / app.stage.scale.x,
+          y: (mousePos.y - app.stage.position.y) / app.stage.scale.y,
+        };
+
+        app.stage.scale.set(newScale);
+
+        const newScreenPos = {
+          x: worldPos.x * newScale + app.stage.position.x,
+          y: worldPos.y * newScale + app.stage.position.y,
+        };
+
+        app.stage.position.set(
+          Math.min(0, Math.max(app.stage.position.x - (newScreenPos.x - mousePos.x), app.screen.width - app.screen.width * newScale)),
+          Math.min(0, Math.max(app.stage.position.y - (newScreenPos.y - mousePos.y), app.screen.height - app.screen.height * newScale)),
+        );
+      }
+    });
+
+    // Panning functionality
+    let isDragging = false;
+    let startDragPos = { x: 0, y: 0 };
+
+    canvas.value?.addEventListener("mousedown", (event) => {
+      isDragging = true;
```

Comment:
Can we, please, change the cursor to:
- `grab` when the view is zoomed in and `isDragging` is `false`
- `grabbing` when the view is zoomed in and `isDragging` is `true`

?